jetty-websocket/javax-websocket-client/src/main/java/org/eclipse/jetty/websocket/javax/client/JavaxWebSocketClientContainer.java

```diff
@@ -52,33 +53,29 @@
 @ManagedObject("JSR356 Client Container")
 public class JavaxWebSocketClientContainer extends JavaxWebSocketContainer implements javax.websocket.WebSocketContainer
 {
-    protected final WebSocketCoreClient coreClient;
+    protected WebSocketCoreClient coreClient;
+    protected Supplier<WebSocketCoreClient> coreClientFactory;
     private final JavaxWebSocketClientFrameHandlerFactory frameHandlerFactory;
-    private ClassLoader contextClassLoader;
     private DecoratedObjectFactory objectFactory;
     private WebSocketExtensionRegistry extensionRegistry;
 
     public JavaxWebSocketClientContainer()
     {
-        this(new WebSocketCoreClient());
-        this.coreClient.getHttpClient().setName("Javax-WebSocketClient@" + Integer.toHexString(this.coreClient.getHttpClient().hashCode()));
-        // We created WebSocketCoreClient, let lifecycle be managed by us
-        addManaged(coreClient);
+        this(() -> new WebSocketCoreClient());
     }
 
-    public JavaxWebSocketClientContainer(HttpClient httpClient)
+    public JavaxWebSocketClientContainer(Supplier<WebSocketCoreClient> coreClientFactory)
     {
-        this(new WebSocketCoreClient(httpClient));
-        // We created WebSocketCoreClient, let lifecycle be managed by us
-        addManaged(coreClient);
+        this((WebSocketCoreClient)null);
+        this.coreClientFactory = coreClientFactory;
+        this.addBean(coreClientFactory);
```

Comment:
If the Supplier is outside of `org.eclipse.jetty.websocket` then we cannot addBean here.
It could be a common WebSocketCoreClient, or an common HttpClient.
Only the no-args constructor can use addBean here.

Reply:
We can use addBean here as it will add it as an auto bean.  If it is external, then it will be started and just become an unmanaged bean.  If we have to start it (even if it is external) then we should stop it, so it becomes managed bean.